torchvision/models/swin_transformer.py

```diff
@@ -92,26 +136,28 @@ def shifted_window_attention(
         window_size (List[int]): Window size.
         num_heads (int): Number of attention heads.
         shift_size (List[int]): Shift size for shifted window attention.
-        attention_dropout (float): Dropout ratio of attention weight. Default: 0.0.
-        dropout (float): Dropout ratio of output. Default: 0.0.
         qkv_bias (Tensor[out_dim], optional): The bias tensor of query, key, value. Default: None.
         proj_bias (Tensor[out_dim], optional): The bias tensor of projection. Default: None.
+        logit_scale (Tensor[out_dim], optional): Logit scale of cosine attention for Swin Transformer V2. Default: None.
     Returns:
         Tensor[N, H, W, C]: The output tensor after shifted window attention.
     """
     B, H, W, C = input.shape
+
+    # If window size is larger than feature size, there is no need to shift window
+    if window_size[0] >= H:
+        shift_size[0] = 0
+        window_size[0] = H
```

Comment:
As discussed on chat, since this might introduce issue because relative_bias_index size is based on window_size, we should revert this and not having dynamic window_size for now.